### ISSUE TITLE
Is this Project Alive? (and json identifier transformations)

### ISSUE THREAD

**Issue by snunez1** [hankhero/cl-json#7]
Does anyone have any utilities for json strings that come back in dotted notation? Especially when using REST to configure servers I see the majority of properties come back 'Java style', e.g. (server.port 8080). The default decoder translates this to SERVER--PORT.

Anyone have a utilities for dotted property notation and underscores?

**Comment by rpgoldman** [comment]
I'd suggest you edit this issue so that it is an issue (i.e., focus on the identifier transformations). For liveness and management issues, try cl-json-devel@common-lisp.net.  It would be useful to figure out if that forum is still live.

**Comment by rpgoldman** [comment]
I note that underscores are translated to pairs of hyphens according to the CL-JSON docs (http://common-lisp.net/project/cl-json/#CAMEL-CASE-TRANSLATION).  What do you think is the right way to translate these Java style identifiers? Preserve the period?  What about case?
I suppose arguably the right thing would have been to simply translate the identifiers into mixed-case symbols, which are a little cumbersome to type, but wouldn't be lossy....

**Comment by snunez1** [comment]
I¹d say that underscores, and full-stops (periods) ought to be translated
into hyphens. They¹re almost always used as Œseparators¹ of sorts in
variable naming, so naturally map to hyphens in LISP. I¹m talking about
JSON->LISP here. For the reverse, perhaps a configurable switch: hyphens to
either Œ.¹, Œ_¹ or camel case.

From:  rpgoldman notifications@github.com
Reply-To:  hankhero/cl-json
<reply+i-37102980-c3543588e0c5acfd21e3d67d0df3c666a185832a-7775472@reply.git
hub.com>
Date:  Tuesday, July 8, 2014 at 13:21
To:  hankhero/cl-json cl-json@noreply.github.com
Cc:  Steve Nunez snunez@hortonworks.com
Subject:  Re: [cl-json] Is this Project Alive? (and json identifier
transformations) (#7)

I note that underscores are translated to pairs of hyphens according to the
CL-JSON docs 
(http://common-lisp.net/project/cl-json/#CAMEL-CASE-TRANSLATION).  What do
you think is the right way to translate these Java style identifiers?
Preserve the period?  What about case?
I suppose arguably the right thing would have been to simply translate the
identifiers into mixed-case symbols, which are a little cumbersome to type,
but wouldn't be lossy....

‹
Reply to this email directly or view it on GitHub
https://github.com/hankhero/cl-json/issues/7#issuecomment-48394793 .

## 

CONFIDENTIALITY NOTICE
NOTICE: This message is intended for the use of the individual or entity to 
which it is addressed and may contain information that is confidential, 
privileged and exempt from disclosure under applicable law. If the reader 
of this message is not the intended recipient, you are hereby notified that 
any printing, copying, dissemination, distribution, disclosure or 
forwarding of this communication is strictly prohibited. If you have 
received this communication in error, please contact the sender immediately 
and delete it from your system. Thank You.

**Comment by rpgoldman** [comment]
Looks like you can define your own function, and bind `*json-identifier-name-to-lisp*` to that function, so maybe this isn't a bug?

**Comment by snunez1** [comment]
No, it¹s not a bug, and I fixed it exactly as you described. I supposed this
is more a suggestion for improvement. Handling those three cases (dot,
underscore and camel case) would cover 90% or more of the cases Œout of the
box¹.

The learning curve was rather steep to make such a minor change, so the
suggestion really is: provide defaults that handle those three cases, or
perhaps add an example to the FAQ.

From:  rpgoldman notifications@github.com
Reply-To:  hankhero/cl-json
<reply+i-37102980-c3543588e0c5acfd21e3d67d0df3c666a185832a-7775472@reply.git
hub.com>
Date:  Tuesday, July 8, 2014 at 13:33
To:  hankhero/cl-json cl-json@noreply.github.com
Cc:  Steve Nunez snunez@hortonworks.com
Subject:  Re: [cl-json] Is this Project Alive? (and json identifier
transformations) (#7)

Looks like you can define your own function, and bind
_json-identifier-name-to-lisp_ to that function, so maybe this isn't a bug?

‹
Reply to this email directly or view it on GitHub
https://github.com/hankhero/cl-json/issues/7#issuecomment-48395979 .

## 

CONFIDENTIALITY NOTICE
NOTICE: This message is intended for the use of the individual or entity to 
which it is addressed and may contain information that is confidential, 
privileged and exempt from disclosure under applicable law. If the reader 
of this message is not the intended recipient, you are hereby notified that 
any printing, copying, dissemination, distribution, disclosure or 
forwarding of this communication is strictly prohibited. If you have 
received this communication in error, please contact the sender immediately 
and delete it from your system. Thank You.
